src/AbstractObjects.php

```diff
@@ -112,77 +114,20 @@ protected function processLanguages( $translations ) {
 	 * @return void
 	 */
 	protected function processTranslations( $translations ) {
-		global $wpdb;
-
-		$terms = [];
-
-		foreach ( array_keys( $translations ) as $name ) {
-			$terms[] = $wpdb->prepare( '(%s, %s)', $name, $name );
-		}
-
-		$terms = array_unique( $terms );
-
-		// Insert terms.
-		if ( ! empty( $terms ) ) {
-			$wpdb->query( "INSERT INTO $wpdb->terms (slug, name) VALUES " . implode( ',', $terms ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-		}
+		$translated_object = false;
 
-		// Get all terms with their term_id.
-		$terms = $wpdb->get_results(
-			sprintf(
-				"SELECT term_id, slug FROM $wpdb->terms WHERE slug IN ( '%s' )",
-				implode( "', '", esc_sql( array_keys( $translations ) ) )
-			)
-		);
-
-		$tts = [];
-
-		foreach ( $terms as $term ) {
-			$tts[] = $wpdb->prepare(
-				'(%d, %s, %s, %d)',
-				$term->term_id,
-				$this->getTranslationTaxonomy(),
-				serialize( $translations[ $term->slug ] ), // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
-				count( $translations[ $term->slug ] )
-			);
-		}
-
-		$tts = array_unique( $tts );
-
-		// Insert term taxonomy part of terms.
-		if ( ! empty( $tts ) ) {
-			$wpdb->query( "INSERT INTO $wpdb->term_taxonomy (term_id, taxonomy, description, count) VALUES " . implode( ',', $tts ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-		}
-
-		unset( $terms, $tts ); // Free some memory.
-
-		// Get all terms with their term taxonomy id.
-		$terms = $wpdb->get_results(
-			sprintf(
-				"SELECT tt.term_taxonomy_id, t.slug FROM $wpdb->terms AS t
-				INNER JOIN $wpdb->term_taxonomy AS tt ON t.term_id = tt.term_id
-				WHERE tt.taxonomy = '%s'
-				AND t.slug IN ( '%s' )",
-				esc_sql( $this->getTranslationTaxonomy() ),
-				implode( "', '", esc_sql( array_keys( $translations ) ) )
-			)
-		);
-
-		$trs = [];
-
-		if ( is_array( $terms ) ) {
-			foreach ( $terms as $term ) {
-				foreach ( $translations[ $term->slug ] as $object_id ) {
-					$trs[] = sprintf( '(%d, %d)', (int) $object_id, (int) $term->term_taxonomy_id );
-				}
+		foreach ( PLL()->model->translatable_objects as $translatable_object ) {
+			if ( $translatable_object->get_tax_translations() === $this->getTranslationTaxonomy() ) {
+				$translated_object = $translatable_object;
+				break;
 			}
 		}
 
-		$trs = array_unique( $trs );
-
-		// Insert term relationships.
-		if ( ! empty( $trs ) ) {
-			$wpdb->query( "INSERT INTO $wpdb->term_relationships (object_id, term_taxonomy_id) VALUES " . implode( ',', $trs ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		if ( ! $translated_object instanceof PLL_Translated_Object ) {
+			// Uh?
+			return;
 		}
+
+		$translated_object->set_translation_in_mass( $translations );
```

Comment:
I get that `set_translation_in_mass()` is available since 3.4, but the new feature you introduced in https://github.com/polylang/polylang/pull/1664#pullrequestreview-2842743837 won't be available right away.
Given that, should we have a backward compatibility?

Reply:
I asked @Chouby first about this format `pll_wpml_{trid}` and it is not important, it has never been useful so far, so I chose not to constrain to use PLL >= 3.8.
However, I wanted to preserve the format `pll_wpml_{trid}` just in case, because it's a tiny change in PLL, so I made https://github.com/polylang/polylang/pull/1664. It's one of these situations where "it's OK to have it but meh".

But I'm not against setting `WPML_TO_POLYLANG_MIN_PLL_VERSION` to `3.8` if you or @Chouby think we should.

Reply:
Setting to 3.8 might to high IMHO.

Also, I'm the kind of person to think "If it's not used, let's get rid of it". So I would purely ignore the change then.

Reply:
Nothing to do then.